File: ios/SwiftKotlination/SwiftKotlination/Sources/Features/TopStories/Models/Story.swift
import ObjectMapper

struct Story: ImmutableMappable {
    let section: String
    let subsection: String
    let title: String
    let abstract: String
    let url: String
    let byline: String
    
    init(map: Map) throws {
        section         = try map.value("section")
        subsection      = try map.value("subsection")
        title           = try map.value("title")
        abstract        = try map.value("abstract")
        url             = try map.value("url")
        byline          = try map.value("byline")
    }
    
    func mapping(map: Map) {
        section         >>> map["section"]
        subsection      >>> map["subsection"]
        title           >>> map["title"]
        abstract        >>> map["abstract"]
        url             >>> map["url"]
        byline          >>> map["byline"]
    }
}

File: android/SwiftKotlination/app/src/main/java/fr/jhandguy/swiftkotlination/features/story/model/Story.kt
package fr.jhandguy.swiftkotlination.features.story.model

import fr.jhandguy.swiftkotlination.model.Multimedia
import kotlinx.serialization.Serializable

@Serializable
data class Story(
        val section: String = "",
        val subsection: String = "",
        val title: String = "",
        val abstract: String = "",
        val url: String = "",
        val byline: String = "",
        val multimedia: List<Multimedia> = emptyList())